### PR TITLE
888 Extend description for date parameters

### LINKED ISSUE #888: Extend description for date parameters
For parameters which expect a date a reference to `derive_vars_dt()` should be added.

### PR DESCRIPTION
**Pull request by wqrena**
Close #888.

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Review by bundfussr — CHANGES_REQUESTED** [review]
Thanks @wqrena . Looks good.

Please wrap the new lines to avoid the lintr warnings.

**Comment by wqrena** [comment]
> 
@bundfussr Thanks for review.
"re-flow comment" is used to wrap the comments automatically. lintr is fine now.

File: R/derive_var_ontrtfl.R [hunk 1]
```diff
@@ -1,81 +1,91 @@
...
+#' date or start/stop dates fall between 2 dates. The following logic is used to
+#' assign on-treatment = `"Y"`: 1. `start_date` is missing and
+#' `ref_start_date`is non-missing 2. No timepoint filter is provided
+#' (`filter_pre_timepoint`) and both `start_date` and `ref_start_date` are
+#' non-missing and `start_date` = `ref_start_date` 3. A timepoint is provided
+#' (`filter_pre_timepoint`) and both `start_date` and `ref_start_date` are
+#' non-missing and `start_date = ref_start_date` and the filter provided in
+#' `filter_pre_timepoint` is not true. 4. `ref_end_date` is not provided and
+#' `ref_start_date < start_date` 5. `ref_end_date` is provided and
+#' `ref_start_date < start_date` <= `ref_end_date + ref_end_window`.
```
Comment (bundfussr) [inline R/derive_var_ontrtfl.R:71]: This is no longer rendered properly.

Comment (wqrena) [reply R/derive_var_ontrtfl.R]: It seems that using `Code|Reflow Comment` to wrap the comments automatically somehow will wrap sentences even though an `enter` key is there. So after wrap the comments automatically, I manually added `enter` key between numbered sentences. Now the rendering looks fine to me.

**Review by bundfussr — APPROVED** [review]

**Pull request merged**